nixos/modules/services/monitoring/prometheus/exporters/modemmanager.nix

```diff
@@ -0,0 +1,35 @@
+{ config, lib, pkgs, options }:
+
+with lib;
+
+let
+  cfg = config.services.prometheus.exporters.modemmanager;
+in
+{
+  port = 9539;
+  extraOpts = {
+    refreshRate = mkOption {
+      type = types.str;
+      default = "5s";
+      description = ''
+        How frequently ModemManager will refresh the extended signal quality
+        information for each modem. The duration should be specified in seconds
+        ("5s"), minutes ("1m"), or hours ("1h").
+      '';
+    };
+  };
+  serviceOpts = {
+    serviceConfig = {
+      # Required in order to authenticate with ModemManager via D-Bus.
+      DynamicUser = false;
+      # TODO: how should this be plumbed in?
+      # Group = "networkmanager";
```

Comment:
It just occurred to me that adding the exporter's user to the "networkmanager" group is required in order to call D-Bus methods on the modems, but this wouldn't create an error in CI since there are no modems!

How should this situation be handled? I can't enforce a default for the exporter group setting due to a conflict with the default options attributes for each exporter.

Reply:
Would `SupplementalGroups` maybe help here? (see `man 5 systemd.exec`)

Reply:
Updated to use SupplementaryGroups. Thanks for the suggestion!

Reply:
Oh good. Just to be sure, would it be enough to use a dynamically allocated user instead with `DynamicUser = true`, while also setting `Group = "networkmanager"`?
The test still seems to work, but as you mentioned, this would not fail without any modems.

Reply:
I think you're right and we can go back to using a dynamic user, I'll update the PR.